clang/test/CodeGenSYCL/regcall-cc-tuple-test.cpp

```diff
@@ -0,0 +1,32 @@
+// RUN:  %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown -disable-llvm-passes -O0 -opaque-pointers -emit-llvm %s -o - | FileCheck %s
+//
+// Check that we use vector ABI in the arguments and return type for a __regcall esimd function with std::tuple
+template <class T, int N> using raw_vector = T[N];
+
+template <class T, int N>
+struct simd {
+  raw_vector<T, N> val;
+};
+
+using T1 = simd<double, 5>;
+using T2 = simd<int, 20>;
+
+namespace std {
+template< class... Types >
+class tuple {
+  // Hardcode this as to not actually implement std::tuple
+  T1 one;
+  T2 two;
+};
+}
+using MyTuple = std::tuple<T1, T2>;
+// CHECK: define dso_local x86_regcallcc <30 x i32> @_Z16__regcall3__funcSt5tupleIJ4simdIdLi5EES0_IiLi20EEEE(<30 x i32> 
```

Comment:
Is it valid to have `x86_regcallcc` calling convention with spir triple?

Reply:
Konst added [these documents](https://github.com/intel/llvm/commit/59dd38795c7557a779ca38cd27df21a40c7b3c45), but if I recall correctly, they don't cover new calling convention.